dotcom-rendering/src/lib/articleCount.ts

```diff
@@ -43,7 +41,10 @@ export const getArticleCounts = async (
 		.filter((tag) => ['tone', 'keyword'].includes(tag.type.toLowerCase()))
 		.map((tag) => tag.id);
 
-	if (!window.guardian.weeklyArticleCount) {
+	if (
+		!window.guardian.weeklyArticleCount &&
+		!(await hasOptedOutOfWeeklyArticleCount())
```

Comment:
I think this breaks the rule in the comment above:
"hasOptedOut needs to be done before we check if articleCount is set in the window"

I wonder if it's best to separate the maintenance of these two local storage items into separate functions.
i.e. split into `getDailyArticleCounts` and `getWeeklyArticleCounts`.
Then, places which require consented article counts must use `getWeeklyArticleCounts` only.